From db466d8594046dca6a786543b33a8c524f8e4037 Mon Sep 17 00:00:00 2001
From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Thu, 25 Apr 2019 00:39:04 +0200
Subject: [PATCH 15/29] network: fix ListenPort= in [WireGuard] section

This fixes a bug introduced by f1368a333e5e08575f0b45dfe41e936b106a8627.

Fixes #12377.

(cherry picked from commit a62b7bb79e9a2aa683624c32cde1c756d8466fb4)
---
 src/network/netdev/wireguard.c | 19 ++++++++++---------
 1 file changed, 10 insertions(+), 9 deletions(-)

diff --git a/src/network/netdev/wireguard.c b/src/network/netdev/wireguard.c
index 7959c1c01f..6babb7950a 100644
--- a/src/network/netdev/wireguard.c
+++ b/src/network/netdev/wireguard.c
@@ -453,22 +453,23 @@ int config_parse_wireguard_listen_port(
                 void *userdata) {
 
         uint16_t *s = data;
-        uint16_t port = 0;
         int r;
 
         assert(rvalue);
         assert(data);
 
-        if (!streq(rvalue, "auto")) {
-                r = parse_ip_port(rvalue, s);
-                if (r < 0) {
-                        log_syntax(unit, LOG_ERR, filename, line, r,
-                                   "Invalid port specification, ignoring assignment: %s", rvalue);
-                        return 0;
-                }
+        if (isempty(rvalue) || streq(rvalue, "auto")) {
+                *s = 0;
+                return 0;
+        }
+
+        r = parse_ip_port(rvalue, s);
+        if (r < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, r,
+                           "Invalid port specification, ignoring assignment: %s", rvalue);
+                return 0;
         }
 
-        *s = port;
         return 0;
 }
 
-- 
2.21.0

